regex/Regex/VM/Basic.lean

```diff
@@ -41,30 +41,32 @@ def εClosure {bufferSize : Nat} (nfa : NFA) (wf : nfa.WellFormed) (it : Iterato
     if state ∈ next.states then
       εClosure nfa wf it matched next stack'
     else
-      let states' := next.states.insert state
```

Comment:
Lean doesn't recognize that we don't use `next` (but uses `next.states` and `next.updates` individually), considering `next.states` have two reference counts.